### ISSUE TITLE
[CI] DockerTests test085EnvironmentVariablesAreRespectedUnderDockerExec failing on Fedora 32

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#66656]
This docker test seems to fail on occasion specifically on Fedora 32:

https://gradle-enterprise.elastic.co/s/zjf65ovw657li/tests/:qa:os:destructiveDistroTest.default-docker_ubi/org.elasticsearch.packaging.test.DockerTests/test085EnvironmentVariablesAreRespectedUnderDockerExec#1

```
org.elasticsearch.packaging.test.DockerTests > test010Install FAILED
    java.lang.RuntimeException: Command was not successful: [bash -c docker exec --user elasticsearch:root --tty 2e1612134c884d03d13f7a711a416b26db7e2d0c5643c19aa9a998305f313fe1 stat --format="%U %G %A" /usr/share/elasticsearch/config/elasticsearch.keystore]
       result: exitCode = [1] stdout = [stat: cannot stat '/usr/share/elasticsearch/config/elasticsearch.keystore': No such file or directory] stderr = []
        at __randomizedtesting.SeedInfo.seed([528C4E8802F598B8:76D3EE1B6058C7E0]:0)
        at org.elasticsearch.packaging.util.Shell.runScript(Shell.java:158)
        at org.elasticsearch.packaging.util.Shell.run(Shell.java:84)
        at org.elasticsearch.packaging.util.Docker.assertPermissionsAndOwnership(Docker.java:372)
        at org.elasticsearch.packaging.util.Docker.verifyOssInstallation(Docker.java:426)
        at org.elasticsearch.packaging.util.Docker.verifyContainerInstallation(Docker.java:408)
        at org.elasticsearch.packaging.test.DockerTests.test010Install(DockerTests.java:114)
```

**Comment by costin** [comment]
Another instance here: https://gradle-enterprise.elastic.co/s/s22zq3hzut2tg

**Comment by ywangd** [comment]
Another one, but on ubuntu-18.04: https://gradle-enterprise.elastic.co/s/jkuoddwnkjkzs

This test [fails](https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=Australia/Melbourne&tests.container=org.elasticsearch.packaging.test.DockerTests&tests.sortField=FAILED&tests.test=test085EnvironmentVariablesAreRespectedUnderDockerExec&tests.unstableOnly=true) almost everyday since 2020-12-30 and the OS are fedora 32, debian 9 and ubuntu 18.04.

**Comment by mark-vieira** [comment]
@pugnascotia is this at all similar to these failures?

https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P28D&search.tags=not:pull-request&search.timeZoneId=America/Los_Angeles&tests.container=org.elasticsearch.packaging.test.DockerTests&tests.disabledDistributions=WyJvdXRjb21lOnNraXBwZWQiLCJvdXRjb21lOnBhc3NlZCJd&tests.sortField=FAILED&tests.test=test010Install&tests.unstableOnly=true

**Comment by pugnascotia** [comment]
@mark-vieira Yes, though I'm a little dismayed to see another failure.

I think we (or rather, I) need to change this so that the keystore permissions are explicitly checked in a dedicated test, and remove the check from `verifyOssInstallation()` because it's too easy to cause problems.

**Comment by davidkyle** [comment]
There was a number of failures on `DockerTests test010Install` over the weekend with the same missing file exception:

`stat: cannot stat '/usr/share/elasticsearch/config/elasticsearch.keystore': No such file or directory`

https://gradle-enterprise.elastic.co/s/wpnezgpi6h5fu
https://gradle-enterprise.elastic.co/s/c3ah7462xlnz6

These were all on 7.x using the successful commit 27744169228f43592f94cd313606fa9718760362 which contains the backport of the fix in abc645424bbb36273108474d04b219469f44f30d

**Comment by pugnascotia** [comment]
Both those build scans indicate `7.11`, not `7.x`. I'll backport the fix to `7.11` as well, which I should have done in the first place TBH.

**Comment by pugnascotia** [comment]
OK, backported to `7.11` in 019390412fe.
